packages/synapse-react-client/src/lib/utils/SynapseClient.ts

```diff
@@ -2859,11 +2859,12 @@ export const searchAccessRequirements = (
  * @param {string} requirementId id of entity to lookup
  * @returns {AccessRequirementStatus}
  */
-export const getAccessRequirementStatus = (
-  accessToken: string | undefined,
-  requirementId: string | number,
-): Promise<AccessRequirementStatus | ManagedACTAccessRequirementStatus> => {
-  return doGet(
+export function getAccessRequirementStatus<
```

Comment:
Add optional generic parameter so we can (safely) assert the return type if we know what to expect . For example, if we're fetching the status of a `ManagedACTAccessRequirement`, then we know we're going to get a `ManagedACTAccessRequirementStatus`